addons/web/static/src/js/core/dom.js

```diff
@@ -342,15 +343,14 @@ var dom = {
             // a 'real' debounce creation useless. Also, during the debouncing
             // part, the button is disabled without any visual effect.
             $button.addClass('o_debounce_disabled');
+            this.trigger_up('btn_spin_request', {btn: $button[0], spin: true});
+            const always = () => {
+                $button.removeClass('disabled').prop('disabled', false);
+                this.trigger_up('btn_spin_request', {btn: $button[0]});
+            };
             Promise.resolve(dom.DEBOUNCE && concurrency.delay(dom.DEBOUNCE)).then(function () {
-                $button.addClass('disabled').prop('disabled', true);
```

Comment:
> Imo the change in makeButtonHandler() (cc @qsm-odoo) is debatable, I am not against it but to check if that's something we want, eg it makes Add to Cart btn load?

Should not be the way you added it: the spin should only appear after the debounce part (so here where the disabled class was added). That way I think this is less debatable: the user clicks on the button, it is "disabled" thanks to the addition of the `o_debounce_disabled` class, but the user cannot see anything. Then after a delay, a spin appear... but only if the action took more than a few hundred a ms (so only for long RPC actions).

Reply:
Ok to me